### ISSUE TITLE
RAPTOR: raptor worker fails with set of errors

### ISSUE THREAD

**Issue by AymenFJA** [radical-cybertools/radical.pilot#2827]
This issue can be broken into 3 issues or so to fit our `sprint` approach. The issue happens on both `local.locahost` and `ornl.sunmit` and it is blocking https://github.com/radical-cybertools/other_activities/issues/6 on summit.

Error1: `None MPI communicator.`
```
Traceback (most recent call last):
  File "/home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor/lib/python3.8/site-packages/radical/pilot/raptor/worker.py", line 321, in _dispatch_func
    val = to_call(*args, **kwargs)
  File "./raptor_master.py", line 34, in func_mpi
    print('hello %d/%d: %s' % (comm.rank, comm.size, msg))
AttributeError: 'NoneType' object has no attribute 'rank'
```

Error2:  `hello_mpi` no longer exists, and it should be replaced with `my_hello.` (I prefer the old name)
```
Traceback (most recent call last):
  File "/home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor/lib/python3.8/site-packages/radical/pilot/raptor/worker.py", line 264, in _dispatch_func
    to_call, _args, _kwargs = PythonTask.get_func_attr(func)
  File "/home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor/lib/python3.8/site-packages/radical/pilot/pytask.py", line 63, in get_func_attr
    pytask = deserialize_bson(bson_obj)
  File "/home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor/lib/python3.8/site-packages/radical/pilot/utils/serializer.py", line 101, in deserialize_bson
    return pickle.loads(codecs.decode(obj.encode(), "base64"))
_pickle.UnpicklingError: unpickling stack underflow
1675466812.867 : master.000000.worker.0000 : 12105 : 139973129783040 : ERROR    : no hello_mpi in
``` 

Error3: Deserialization error which is mainly due to wrong identification of function type (serialized function or not on the worker level)
```
1675466819.091 : master.000000.worker.0000 : 12198 : 139925236201216 : DEBUG    : func serialized: 5: hello
1675466819.092 : master.000000.worker.0000 : 12198 : 139925236201216 : DEBUG    : func serialized: 5: hello
1675466819.092 : master.000000.worker.0000 : 12198 : 139925236201216 : ERROR    : function is not a PythonTask [task.call.c.000000]
Traceback (most recent call last):
  File "/usr/lib/python3.8/encodings/base64_codec.py", line 19, in base64_decode
    return (base64.decodebytes(input), len(input))
  File "/usr/lib/python3.8/base64.py", line 546, in decodebytes
    return binascii.a2b_base64(s)
binascii.Error: Invalid base64-encoded string: number of data characters (5) cannot be 1 more than a multiple of 4

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor/lib/python3.8/site-packages/radical/pilot/raptor/worker.py", line 264, in _dispatch_func
    to_call, _args, _kwargs = PythonTask.get_func_attr(func)
  File "/home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor/lib/python3.8/site-packages/radical/pilot/pytask.py", line 63, in get_func_attr
    pytask = deserialize_bson(bson_obj)
  File "/home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor/lib/python3.8/site-packages/radical/pilot/utils/serializer.py", line 101, in deserialize_bson
    return pickle.loads(codecs.decode(obj.encode(), "base64"))
binascii.Error: decoding with 'base64' codec failed (Error: Invalid base64-encoded string: number of data characters (5) cannot be 1 more than a multiple of 4)
```

I use `raptor.py` example as is from the `@devel` branch and run it with no modifications.

My `agent/client radical stack` on both machines: 

local:
```
PREFIX    : /home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000//env/rp_named_env.ve_raptor
VERSION   : 3.8
MODULES   : /home/aymen/RADICAL/test_raptor/radical.pilot /home/aymen/ve/raptor/lib/python3.8/site-packages/radical/utils/radical.utils-1.21.0.tar.gz mpi4py apache-libcloud chardet colorama idna msgpack msgpack-python netifaces ntplib parse dill pyzmq regex requests setproctitle urllib3
DEFAULTS  : True
PYTHON    : /home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor/bin/python3 (Python 3.8.10)
PYTHONPATH:
RCT_STACK :
  python               : /home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor/bin/python3
  pythonpath           :
  version              : 3.8.10
  virtualenv           : /home/aymen/radical.pilot.sandbox/rp.session.surfacebook.aymen.019391.0003/pilot.0000/env/rp_named_env.ve_raptor

  radical.gtod         : 1.20.1
  radical.pilot        : 1.22.0-v1.21.0-1-g37f65b858@devel
  radical.saga         : 1.21.0
  radical.utils        : 1.21.0
```

summit: 

```
PWD       : /gpfs/alpine/scratch/aymen/csc449/radical.pilot.sandbox/rp.session.batch2.aymen.019391.0015/pilot.0000
ENV       : /gpfs/alpine/scratch/aymen/csc449/radical.pilot.sandbox/rp.session.batch2.aymen.019391.0015/pilot.0000//env/rp_named_env.ve_raptor.env
SCRIPT    : /autofs/nccs-svm1_home1/aymen/ve/CYLON/bin/radical-pilot-create-static-ve
PREFIX    : /gpfs/alpine/scratch/aymen/csc449/radical.pilot.sandbox/rp.session.batch2.aymen.019391.0015/pilot.0000//env/rp_named_env.ve_raptor
VERSION   : 3.8
MODULES   : /ccs/home/aymen/RADICAL/CYLON-PIP/radical.pilot /ccs/home/aymen/ve/CYLON/lib/python3.7/site-packages/radical/utils/radical.utils-1.20.1.tar.gz mpi4py apache-libcloud chardet colorama idna msgpack msgpack-python netifaces ntplib parse dill pyzmq regex requests setproctitle urllib3
DEFAULTS  : True
PYTHON    : /gpfs/alpine/scratch/aymen/csc449/radical.pilot.sandbox/rp.session.batch2.aymen.019391.0015/pilot.0000/env/rp_named_env.ve_raptor/bin/python3 (Python 3.8.3)
PYTHONPATH:
RCT_STACK :
  python               : /gpfs/alpine/scratch/aymen/csc449/radical.pilot.sandbox/rp.session.batch2.aymen.019391.0015/pilot.0000/env/rp_named_env.ve_raptor/bin/python3
  pythonpath           :
  version              : 3.8.3
  virtualenv           : /gpfs/alpine/scratch/aymen/csc449/radical.pilot.sandbox/rp.session.batch2.aymen.019391.0015/pilot.0000/env/rp_named_env.ve_raptor

  radical.gtod         : 1.20.1
  radical.pilot        : 1.22.0-v1.21.0-1-g37f65b858@devel
  radical.saga         : 1.21.0
  radical.utils        : 1.20.1

```

**Comment by andre-merzky** [comment]
When you run on `local.localhost`, is that also on summit or on your local linux machine?

**Comment by AymenFJA** [comment]
> When you run on `local.localhost`, is that also on summit or on your local linux machine?

`local.localhost` refers to my local linux machine sorry for the confusion.

**Comment by andre-merzky** [comment]
Thanks Aymen.  Managed to reproduce this now, working on it.

**Comment by mturilli** [comment]
@andre-merzky ping?

**Comment by andre-merzky** [comment]
This has been addressed by #2892 - closing.
